src/backend/marsha/core/lti/utils.py

```diff
@@ -0,0 +1,144 @@
+            "and/or consumer site ({!s}).".format(
+                resource.id, lti.context_id, lti.get_consumer_site().domain
+            )
+        )
+
+    playlist, _ = Playlist.objects.get_or_create(
+        lti_id=lti.context_id,
+        consumer_site=lti.get_consumer_site(),
+        defaults={"title": lti.context_title},
+    )
+
+    return playlist
+
+
+def get_or_create_video(lti):
+    """Get or create the video targeted by the LTI launch request.
+
+    Create the playlist if it does not pre-exist (it can only happen with consumer site scope
+    passports).
+
+    Parameters
+    ----------
+    lti : Type[LTI]
+
+    Raises
+    ------
+    LTIException
+        Exception raised if the request does not contain a context id (playlist).
+
+    Returns
+    -------
+    core.models.video.Video
+        The video instance targeted by the launch url or None.
+
+    """
+    # Make sure LTI verification has run successfully. It raises an LTIException otherwise.
+    lti.verify()
+    try:
+        video = _get_resource(Video, lti)
+    except Video.DoesNotExist:
+        if not lti.is_instructor:
+            return None
+    else:
+        return video
+
+    # Creating the video...
+    # - Get the playlist if it exists or create it
+    playlist = _get_or_create_playlist(Video, lti)
+
+    # Create the video, pointing to the file from the origin video if any
+    return Video.objects.create(
+        pk=lti.resource_id,
+        lti_id=lti.resource_link_id,
+        playlist=playlist,
+        upload_state=PENDING,
+        title=lti.resource_link_title,
+        show_download=lti.get_consumer_site().video_show_download_default,
+    )
+
+
+def get_or_create_file(lti):
+    """Get or create the file targeted by the LTI launch request.
+
+    Create the playlist if it does not pre-exist (it can only happen with consumer site scope
+    passports).
+
+    Parameters
+    ----------
+    lti : Type[LTI]
+
+    Raises
+    ------
+    LTIException
+        Exception raised if the request does not contain a context id (playlist).
+
+    Returns
+    -------
+    core.models.video.Video
+        The video instance targeted by the launch url or None.
+
+    """
+    # Make sure LTI verification has run successfully. It raises an LTIException otherwise.
+    lti.verify()
+    try:
+        file = _get_resource(File, lti)
+    except File.DoesNotExist:
+        if not lti.is_instructor:
+            return None
+    else:
+        return file
+
+    # Creating the video...
+    # - Get the playlist if it exists or create it
+    playlist = _get_or_create_playlist(File, lti)
+
+    # Create the video, pointing to the file from the origin video if any
+    return File.objects.create(
+        pk=lti.resource_id,
+        lti_id=lti.resource_link_id,
+        playlist=playlist,
+        upload_state=PENDING,
+        title=lti.resource_link_title,
+    )
```

Comment:
The code in `get_or_create_video` is the same as the code in `get_or_create_file` so you could probably factorize it further inside `_get_resource`?

Reply:
Almost all the code is the same. Only the creation is different. I didn't put the `show_download` field in the `File` model, maybe  i should ?

Reply:
I will add this field. We will need it I think.